src/services/YotiCallbackProcessor.ts

```diff
@@ -193,6 +195,66 @@ export class YotiCallbackProcessor {
   			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Failed to send to IPV Core", { shouldThrow: true });
   		}
 
+  		// Document type objects to pass into TxMA event F2F_CRI_VC_ISSUED
```

Comment:
nit: Can maybe replace the below with a switch case to help readability?

Reply:
```suggestion
  		let docName;
let documentInfo;

switch (documentFields.document_type) {
  case DocumentTypes.PASSPORT:
    docName = DocumentNames.PASSPORT;
    documentInfo = [{
      documentType: documentFields.document_type,
      documentNumber: documentFields.document_number,
      expiryDate: documentFields.expiration_date,
      icaoIssuerCode: documentFields.issuing_country,
    }];
    break;
  case DocumentTypes.RESIDENCE_PERMIT:
    docName = DocumentNames.RESIDENCE_PERMIT;
    documentInfo = [{
      documentType: documentFields.document_type,
      documentNumber: documentFields.document_number,
      expiryDate: documentFields.expiration_date,
      issueDate: documentFields.date_of_issue,
      icaoIssuerCode: documentFields.issuing_country,
    }];
    break;
  case DocumentTypes.DRIVING_LICENCE:
    docName = DocumentNames.DRIVING_LICENCE;
    documentInfo = [{
      documentType: documentFields.document_type,
      personalNumber: documentFields.document_number,
      expiryDate: documentFields.expiration_date,
      issuingCountry: documentFields.issuing_country,
    }];

    if (documentFields.issuing_country !== "GBR") {
      documentInfo[0].issuedBy = documentFields.place_of_issue;
      documentInfo[0].issueDate = documentFields.date_of_issue;
    } else {
      documentInfo[0].issuedBy = documentFields.issuing_authority;
      documentInfo[0].issueDate = documentFields.date_of_issue;
      documentInfo[0].fullAddress = documentFields.structured_postal_address.formatted_address;
    }
    break;
  case DocumentTypes.NATIONAL_ID:
    docName = DocumentNames.NATIONAL_ID;
    documentInfo = [{
      documentType: documentFields.document_type,
      documentNumber: documentFields.document_number,
      expiryDate: documentFields.expiration_date,
      issueDate: documentFields.date_of_issue,
      icaoIssuerCode: documentFields.issuing_country,
    }];
    break;
  default:
    const errorMessage = `Unable to find document type ${documentFields.document_type}`;
    this.logger.error({ message: errorMessage, messageCode: MessageCodes.INVALID_DOCUMENT_TYPE });
    throw new AppError(HttpCodesEnum.SERVER_ERROR, "Unknown document type");
}
```

Reply:
Found a way to refactor it with Richa!